### PR TITLE
[DX-413][DX-423] StyleTag + oc:cssDidMount & oc:componentDidMount events

### PR DESCRIPTION
**Pull request by nickbalestra**
- Moved back to css containing in template under a style tag
- As soon as the style tag get mounted, fire an oc event `oc:cssDidMount` with the css content in case client-side ones need to handle it.
- renamed `oc:mounted` to `oc:componentDidMount` for consistency

### CODE REVIEW

**Comment by themoch** [comment]
👍

File: packages/oc-template-react-compiler/lib/viewTemplate.js [hunk 1]
```diff
@@ -9,11 +9,12 @@ const viewTemplate = ({
   var staticPath = model.reactComponent.props._staticPath;
   var props = JSON.stringify(model.reactComponent.props);
   return '<div id="${reactRoot}" class="${reactRoot}">' + modelHTML + '</div>' +
+    '<style>${css}</style>' +
     '<script>' +
     'window.oc = window.oc || {};' +
     'oc.cmd = oc.cmd || [];' +
     'oc.cmd.push(function(oc){' +
-      'oc.addStylesToHead(\\'${css}\\');' +
+      'oc.events.fire(\\'oc:cssDidMount\\', \\'${css}\\');' +
```
Comment (matteofigus) [inline packages/oc-template-react-compiler/lib/viewTemplate.js]: I think we had an issue for when (in dev mode) the css has comments. Have we addressed that here? I don't see any sanitisation for that string.

Comment (nickbalestra) [reply packages/oc-template-react-compiler/lib/viewTemplate.js]: I've checked by running dev and I've added a snapshot for the string passed in the client-side acceptance test. So wasn't able yet to reproduce it, let's publish this as 5.0.0 and if @themoch will report that this still breaks in the case of comments within the css, then we can try to nail it further.

Comment (themoch) [reply packages/oc-template-react-compiler/lib/viewTemplate.js]: I can pull this branch and test locally if that helps before a merge.

Comment (matteofigus) [reply packages/oc-template-react-compiler/lib/viewTemplate.js]: That would be great @themoch

**Comment by matteofigus** [comment]
@nickbalestra can you fix the snapshot conflicts?

**Comment by themoch** [comment]
Okay, will do as soon as I can, in a few hours.

File: packages/oc-template-react-compiler/lib/viewTemplate.js (hunk 1 above)
Comment (themoch) [reply packages/oc-template-react-compiler/lib/viewTemplate.js]: tested locally looks good!

**Comment by nickbalestra** [comment]
@matteofigus rebased

**Comment by matteofigus** [comment]
Published v5 @nickbalestra

**Pull request merged**